pipeline/scripts/hmdb_part.R

```diff
@@ -13,93 +13,113 @@ outdir <- cmd_args[1]
 scanmode <- cmd_args[2]
 db <- cmd_args[3]
 ppm <- as.numeric(cmd_args[4])
+stdrun  <- cmd_args[5] # "yes"
+std_loc <- cmd_args[6] # "/hpc/dbg_mz/development/HMDB_stukken/DIMS/hmdb_preparts/" 
 
 # Cut up entire HMDB into small parts based on the new binning/breaks 
 
-load(db)
 load(paste(outdir, "breaks.fwhm.RData", sep = "/"))
 outdir_hmdb <- paste(outdir, "hmdb_part", sep = "/")
 dir.create(outdir_hmdb, showWarnings = FALSE)
 
-if (scanmode=="negative"){
-  label = "MNeg"
-  HMDB_add_iso=HMDB_add_iso.Neg
-} else {
-  label = "Mpos"
-  HMDB_add_iso=HMDB_add_iso.Pos
-}
+# New: in case of a standard run (m/z 69-606) use external HMDB parts
+min_mz <- round(breaks.fwhm[1])
+max_mz <- round(breaks.fwhm[length(breaks.fwhm)])
+# test if standard mz range is used
+if (stdrun == "yes" & min_mz > 68 & min_mz < 71 & max_mz > 600 & max_mz < 610) {
+  # skip generating HMDB parts
+  use_external_HMDB <- TRUE
+  save(use_external_HMDB, file=paste(outdir, "using_external_HMDB_parts.RData", sep="/"))
+  HMDBstukken <- list.files(std_loc, pattern=scanmode) # only positive or negative files
+  save(HMDBstukken, file=paste(outdir, "HMDBstukken.RData", sep="/"))
+  for (h in 1:length(HMDBstukken)) {
```

Comment:
Rename `h` to `index` or `i`?

Reply:
h renamed to hmdb_file_index
for (h in length(hmdb_parts)) replaced by for (hmdb_file_index in hmdb_parts)
to avoid the use of length()

Reply:
In that case the variable should be called `hmdb_file_index` or `hmdb_part`, since it is not an index anymore.